libs/pika/async_mpi/src/mpi_polling.cpp

```diff
@@ -26,14 +26,18 @@
 #include <vector>
 
 namespace pika::mpi::experimental {
+
+    constexpr std::uint32_t max_mpi_streams = 4;
+
     namespace detail {
         // -----------------------------------------------------------------
         /// Holds an MPI_Request and a callback. The callback is intended to be
         /// called when the operation tied to the request handle completes.
         struct request_callback
         {
-            MPI_Request request;
-            request_callback_function_type callback_function;
+            MPI_Request request_;
+            request_callback_function_type callback_function_;
+            stream_index index_{0};
```

Comment:
```suggestion
            stream_index index_{stream_index::default};
```
is maybe more obvious?